src/Optimization/hiopIterate.cpp

```diff
@@ -434,12 +434,12 @@ void hiopIterate::addLinearDampingTermToGrad_x(const double& mu, const double& k
   /*sxl->addLinearDampingTermToGrad(nlp->get_ixl(), nlp->get_ixu(), mu, kappa_d, grad_x);
     sxu->addLinearDampingTermToGrad(nlp->get_ixu(), nlp->get_ixl(), mu, kappa_d, grad_x); */
   //I'll do it in place, in one for loop, to be faster
```

Comment:
I guess the code in lines 444-484, needs to be "ported" and use generic `hiopVector` interface ? @pelesh

Reply:
You are absolutely right @cnpetra. I didn't dare messing up with that part of the code. This is probably the primary reason why we need to use unified virtual memory when running on GPU. We also take performance hit there, because all vectors need to be copied to host memory for this computation to take place.

Reply:
This should be easy to fix, though.